src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
+
+Programmable physical environment has a broad definition and it is used in various fields. The usage included educational use, virtual/augmented reality, factory manufacture, internet of things, building architecture, etc. The term still has a long way for improvement. In the future, programmable physical environment may become more common in the society as a combination of various technology systems.
+
+Embodied Cognition
+------------------
+
+One related theory to programmable physical environment is the embodied cognition. Embodied cognition is the theory that many features of cognition are shaped by aspects of the entire body of the organism. High level mental constructs and performance on various cognitive tasks are all included in the features of cognition. It points out that the cognition of beings is related to the entire aspects of the body. When coming to programmable physical environment, it should remind the designers in what ways can human beings interact with the systems.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "wikiembodiedcognition"
+
+  This wiki page gives a explanation of embodied cognition
+
+Design Insight
+--------------
+
+As a programmable physical environment, the human interation with the system is very important. How human will interact with the system should also be a research topic of programmable physical environment.
+
+Here is a worry about future technologies, where people interact with technologies mostly with their fingers.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "bret2011rant"
+
+  The author gives a deep explanation of definition of a tool: A tool addresses human needs by amplifying human capabilities, and a tool convert what people can do to what people want to do. However, most technologies for now focus more on what they can fit people's needs, but less on how they can fit people's capabilities.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "bret2014humanethought"
+
+  The author holds the opinon that human think not only through part of their body, like visual or auditory, but with the whole parts of body.
+
+In the term programmable physical environment, "environment" is one of the most important part to its definition. Based on the theory of embodied cognition, programmable physical environment should give more focus on interaction between people and the system. For the future design, how people will interact with the environment should be considered more carefully.
```

Comment:
Since you connect "embodied cognition" with the topic here for the first time, you should just cite a primary source on embodied cognition here (not a wikipedia page). Can you remove the Embodied Cognition section above and include the key points here instead?